bootstrapform_jinja/templatetags/bootstrap.py

```diff
@@ -1,110 +1,148 @@
+from django.forms import CheckboxInput, CheckboxSelectMultiple, \
+                         RadioSelect, FileInput
+from django.forms.boundfield import BoundField
 from django.template.loader import get_template
-from django_jinja import library
 from django.utils.safestring import mark_safe
+from django.utils.six import string_types
+from django_jinja import library
+
+from bootstrapform_jinja.config import BOOTSTRAP_COLUMN_COUNT
+
+
+class ClassSet(set):
+    """
+    Extension of a set that will concatenate elements in the set with
+    spaces when the set is converted to string.
+    """
+    def __str__(self):
+        return ' '.join((str(x) for x in self))
+
+
+@library.test
+def checkbox_field(field):
+    """
+    Jinja test to check if a field is a checkbox
+    """
+    return isinstance(field.field.widget, CheckboxInput)
+
+
+@library.test
+def multiple_checkbox_field(field):
+    """
+    Jinja test to check if a field is a multiple value checkbox
+    """
+    return isinstance(field.field.widget, CheckboxSelectMultiple)
 
-from bootstrapform_jinja import config
+
+@library.test
+def radio_field(field):
+    """
+    Jinja test to check if a field is a radio select
+    """
+    return isinstance(field.field.widget, RadioSelect)
+
+
+def add_input_classes(field):
+    """
+    Add form-control to class attribute of the widget of the given field.
+    """
+    if not isinstance(field.field.widget, (CheckboxInput, CheckboxSelectMultiple, RadioSelect, FileInput)):
+        attrs = field.field.widget.attrs
+        attrs['class'] = attrs.get('class', '') + ' form-control'
 
 
 @library.filter
 def bootstrap(element):
-    markup_classes = {'label': '', 'value': '', 'single_value': ''}
-    return render(element, markup_classes)
+    """
+    Render field, form or formset with bootstrap styles
+    """
+    return render(element)
 
 
 @library.filter
 def bootstrap_inline(element):
-    markup_classes = {'label': 'sr-only', 'value': '', 'single_value': ''}
-    return render(element, markup_classes)
+    """
+    Render field, form or formset with bootstrap styles in single line
+    """
+    return render(element, {'label': 'sr-only'})
 
 
 @library.filter
-def bootstrap_horizontal(element, label_cols={}):
+def bootstrap_horizontal(element, label_cols=None, max_columns=None):
+    """
+    Render field, form or formset with bootstrap styles in horizontal layout
+    """
     if not label_cols:
         label_cols = 'col-sm-2 col-lg-2'
+    if isinstance(label_cols, string_types):
+        label_cols = label_cols.split()
+    label_cols = ClassSet(label_cols)
 
-    markup_classes = {
+    if not max_columns:
+        max_columns = BOOTSTRAP_COLUMN_COUNT
+
+    classes = {
         'label': label_cols,
-        'value': '',
-        'single_value': ''
+        'value': ClassSet(),
+        'single_value': ClassSet(),
     }
 
-    for cl in label_cols.split(' '):
-        split_class = cl.split('-')
-
+    for cl in label_cols:
```

Comment:
don't think I wrote this originally, it came from django bootstrap form. I'll believe you that it works if we have some proper tests and coverage.